### PR TITLE
Check if context has changed

### PR DESCRIPTION
**Pull request by ecwillis**

### CODE REVIEW

File: builder [hunk 1]
```diff
@@ -79,6 +80,9 @@ touch $lockfile
 log "fetching from repo"
 git fetch --quiet
 log "checking out ${BRANCH}"
+git checkout ${BRANCH} > /dev/null 2>&1
```
Comment (jgallen23) [inline builder]: what does this line do?

File: builder [hunk 2]
```diff
@@ -101,6 +106,16 @@ if [[ -z "$CONTEXT" ]]; then
   CONTEXT='.'
 fi
 
+#do we even need to build
+
+DIFF=$(git diff --name-only ${LASTHASH} ${COMMIT} ${CONTEXT})
+
+if [[ -z $DIFF ]]
+  echo "No difference in context"
+  exit 1
```
Comment (jgallen23) [inline builder]: I think we should exit 0 - so it doesn't trigger errors

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: builder (hunk 1 above)
Comment (ecwillis) [reply builder]: My thought here is that if you are running the builder a second time from the same dir there is no way to tell if you are on the same you are attempting to merge into to tell what has changed... doing this allows the code below to tell what commit the code is currently on... 

Hopefully that makes sense.

Comment (jgallen23) [reply builder]: no, it doesn't. :p

**Pull request merged**